lib/generators/deimos/active_record_consumer_generator.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require 'rails/generators'
+require 'rails/generators/active_record/migration'
+require 'generators/deimos/schema_class_generator'
+require 'rails/version'
+require 'erb'
+
+# Generates a new Active Record Consumer, as well
+# as all the necessary files and configuration.
+# Streamlines the creation process into a single flow.
+module Deimos
+  module Generators
+    # Generator for ActiveRecordConsumer db migration, Rails model, Consumer, Consumer config
+    # and Deimos Schema class
+    class ActiveRecordConsumerGenerator < Rails::Generators::Base
+      include Rails::Generators::Migration
+      if Rails.version < '4'
+        extend(ActiveRecord::Generators::Migration)
+      else
+        include ActiveRecord::Generators::Migration
+      end
+      source_root File.expand_path('active_record_consumer/templates', __dir__)
+
+      argument :full_schema, desc: 'The fully qualified schema name.', required: true
+      argument :config_path, desc: 'The path to the deimos configuration file, relative to the root directory.', required: false
+
+      no_commands do
+
+        # Creates database migration for creating new table
+        def create_db_migration
+          migration_template('migration.rb', "#{db_migrate_path}/create_#{table_name.underscore}.rb")
+        end
+
+        # Creates Rails Model
+        def create_rails_model
+          template('model.rb', "app/models/#{model_name}.rb")
+        end
+
+        # Creates Kafka Consumer file
+        def create_consumer
+          template('consumer.rb', "app/lib/kafka/models/#{consumer_name.underscore}.rb")
+        end
+
+        # Adds consumer config to config file.
+        # Defaults to deimos.rb if config_path arg is not specified.
+        def create_consumer_config
+          config_file = 'deimos.rb'
+          config_file_path = "#{initializer_path}/#{config_file}"
+          config_file_path = config_path if config_path.present?
+
+          if File.exist?(config_file_path)
```

Comment:
We should validate that the config file exists if it's passed in. If not, we can create it.

Reply:
When I tested it by passing in a config file path that did not already exist, it did go ahead and create that config file.

Reply:
Sorry, that was unclear. We should only create it if no path was specified. If a path was specified and doesn't exist, we should assume that means that they made a mistake and raise an error.

Actually, I'm not sure we need to create it at all. If we want to auto-create a config file, it probably would be in an installer or something, but I don't think we have a use case for that. So maybe I'd just say don't create anything, and throw an error if the file doesn't exist.